gms/versioned_value.hh

```diff
@@ -58,17 +60,17 @@ public:
     // values for ApplicationState.REMOVAL_COORDINATOR
     static constexpr const char* REMOVAL_COORDINATOR = "REMOVER";
 
-    int version;
-    sstring value;
+    version_type version() const noexcept { return _version; };
```

Comment:
this breaks the gdb script at https://github.com/scylladb/scylladb/blob/f51762c72a7ed651695a7bb10409b1c85c55f5e3/scylla-gdb.py#L4046 . might want to s/'version'/'_version'/ there.

Reply:
or, if we want to be more future-proof, we might need to call these accessors in `scylla-gdb.py`.

Reply:
Yes, we should

Reply:
But we need to think about backward compatibility here, since by convention, we use the latest scylla-gdb.py for older releases.

Cc @denesb. What would be the best to approach this?  Define backward-compatible accessor functions for versioned_value et. al.?

Reply:
We cannot rely on calling methods in `scylla-gdb.py` as that doesn't work in coredumps (and such simple accessors are often all inlined in which case it also doesn't work).

The usual way to deal with such changes is:
```
try:
   value = value['value']
except gdb.error:
  value = value['_value'] # backward compatibility with 5.2
```